CRM/Core/DAO.php

```diff
@@ -783,6 +783,13 @@ public function copyValues($params) {
           $this->$dbName = CRM_Core_DAO::serializeField($value, $field['serialize']);
           $allNull = FALSE;
         }
+        // When a single value was entered for a serialized field, it's probably due to sloppy coding.
+        // Folks, always use an array to pass in values for fields containing array data.
+        // Meanwhile, I'll convert it for you. You're welcome.
+        elseif (is_numeric($value) && !empty($field['serialize'])) {
```

Comment:
if we really wanted to discourage this we would add a noisy deprecation - but I suspect you are not too fussed?

Reply:
I was going to do that next as a followup PR. This one has already been difficult enough.